Gravatar/GravatarService.cs

```diff
@@ -1,222 +1,155 @@
 ﻿using System;
-using System.Collections.Generic;
 using System.Diagnostics;
 using System.Drawing;
 using System.Net;
-using System.Text.RegularExpressions;
+using System.Threading.Tasks;
 
 namespace Gravatar
 {
-    /// <summary>
-    /// Services that gravatar provides in order to provide avatars in
-    /// the absence of a user-uploaded image.
-    /// </summary>
-    public enum FallBackService
+    public interface IGravatarService
     {
         /// <summary>
-        /// Return an HTTP 404 respose.
+        /// Removes all cached images.
         /// </summary>
-        None,
+        Task ClearCacheAsync();
 
         /// <summary>
-        /// Return a cartoon-style silhouette.
+        /// Configures the local cache.
         /// </summary>
-        MysteryMan,
+        /// <param name="imageCachePath">The path to the cache folder.</param>
+        /// <param name="cacheDays">The number of days to keep images in the cache.</param>
+        void ConfigureCache(string imageCachePath, int cacheDays);
 
         /// <summary>
-        /// Return a generated monster based on the email hash.
+        /// Loads avatar either from the local cache or from the remote service.
         /// </summary>
-        MonsterId,
+        Task<Image> GetAvatarAsync(string email, int imageSize, DefaultImageType defaultImageType);
 
         /// <summary>
-        /// Return a generated face based on the email hash.
+        /// Removes the avatar from the local cache.
         /// </summary>
-        Wavatar,
-
-        /// <summary>
-        /// Return a geometric pattern based on the email hash.
-        /// </summary>
-        Identicon,
-
-        /// <summary>
-        /// Return an 8-bit-style face based on the email hash.
-        /// </summary>
-        Retro,
+        Task RemoveAvatarAsync(string email);
     }
 
-    /// <summary>
-    /// Specifies the maximum rating of a given gravatar image request.
-    /// </summary>
-    public enum Rating
+    public class GravatarService : IGravatarService
     {
-        /// <summary>
-        /// Suitable for all audiences.
-        /// </summary>
-        G,
+        private IImageCache _cache;
+        private readonly IImageCacheFactory _imageCacheFactory;
 
-        /// <summary>
-        /// May contain rude gestures, provocatively dressed indiviiduals,
-        /// the lesser swear words, or mild violence
-        /// </summary>
-        PG,
 
-        /// <summary>
-        /// May contain such things as harsh profanity, intense violence,
-        /// nudity, or hard drug use.
-        /// </summary>
-        R,
+        public GravatarService(IImageCacheFactory imageCacheFactory)
```

Comment:
Why don't pass an already configured ImageCache instead of ImageCacheFactory?
ImageCacheFactory would not be needed, neither do verification if service is configured.